### DISCUSSION TITLE
[Suggestion] Allow all registered users to be able to view code notes on the site

### DISCUSSION THREAD

**Discussion by Blazekickn** [RetroAchievements/RAWeb#761] [Ideas]
I dont really see why we should make the code notes page dev only.  Jr. devs could benefit from viewing code notes from the site if they are given the link for it. The code notes page doesnt have anything that could do harm as far as i know.

**Comment by Flara-StarLite** [comment]
I like this idea, too. It should make things easier for Jr. Devs if they have all code nodes and their associated memory addresses in one place, so they can refer to it easily while building achievements. It's a nuisance to have to scroll through a drop-down menu each time and pick a number hoping it's the one you're looking for.

> The code notes page doesnt have anything that could do harm as far as i know.

This. There's nothing on that page that non-devs cannot view already from the memory inspector.

**Comment by kdecks** [comment]
I believe the plan is to go in the other direction. Restrict the dev tools entirely from non-devs to deter players using it as a means to cheat games and exploit achievement code.

As for jr. devs v2 will be able to see code notes, fix tickets, make leaderboard, rich presence, etc. The only reason they can't now is the v1 site doesn't have jr devs at all.

**Comment by Blazekickn** [comment]
ok but v2 isnt happenning for a while so id say help make it easier for jr devs now

**Comment by Blazekickn** [comment]
as flara said you can still cheat by using the editor its just less convinient and hurts jr devs. There are plenty of other ways to cheat anyway

**Comment by Blazekickn** [comment]
alternatively work on making a jr dev rank thats just registered copy pasted but with the ability to view the page

**Comment by Ben-Ghazi** [comment]
Super users aren't being used for anything and are already in-between Registered and Developer so it would be perfect if that could be re-purposed into a Jr Dev. rank.

**Comment by kdecks** [comment]
Another potential reason to not expose code notes to general users is notes are an asset of the community, allowing them viewable by increases odds they will easily get scraped. (I might be alone in this view actually mattering, never talked to anyone about it.)

About super users, yeah that'd be ideal. But I wouldn't expect many v1 changes. Any changes made to v1 only delay v2.

**Comment by Blazekickn** [comment]
I already copy a lot of my notes from other places and it already isnt hard to collect all code notes by using the same method the emulators use. The reasons of blocking off code notes from cheating and stealing code notes dont work and hurt jr. devs.

**Comment by Tsearo** [comment]
As a recent Jr-Dev graduate I very much would have liked to have been able to view not only my code notes but code notes for other similar games without having to load up a game and scroll through every address one at a time. It was one of the first things I asked in the Jr-Dev channel since I (accidentally)  had Dev status before I knew that you could look at the entire list of code notes for a game on the site page but didn't know that Jr's couldn't see them :/

**Comment by meleu** [comment]
I like the idea of re-purposing the Super User role to Jr. Developer and then allowing them to view Code Notes and achievement's logic on the site. I will try to implement it over this month (March/2019).

**Comment by Blazekickn** [comment]
I like the idea of repurposing the Super User role a lot more but wasn't expecting it would be done for V1. Maybe we can make it so only Jr. Devs can upload to unofficial also.

**Comment by luchaos** [comment]
some preparations for that went into the permissions branch: https://github.com/RetroAchievements/RAWeb/commit/57574f81444f9be46451ff29d1268bd77e714429
the idea was to repurpose superuser to moderator (and swap order)
now superuser is back below developers, renamed to jr. developer and moderator pushed admins and root one level higher.
https://github.com/RetroAchievements/RAWeb/commit/31324dc94083869c554215031c5e8c991eedcb78
https://github.com/RetroAchievements/RAWeb/commit/b4dbae7b30f34eb494985093022a5672d8022b09
the most important bit there: permission checks have to be done by comparing with class constants - no magic numbers anymore.
then it's easy to lower/raise required permissions with some level of confidence.

**Comment by steffeN-92** [comment]
For me that's a good idea.

My best idea:

Access for only Jr. Dev:
-View code notes on RA page for all sets
-Promote cheevos from unofficial to core for only **his own sets**
-Demote cheevos from core to unofficial for only **his own sets**
-Edit cheevos for only **his own sets**
-Edit/Change rich presences, leaderboards, images, box art, game details for only **his own sets** (except gameicon)
-Resolve/Close/Reopen tickets for only **his own sets**

No access for only Jr. Dev:
-Demote cheevos from core to unofficial from other sets from other developers
-Promote cheevos from unofficial to core from other sets from other developers
-Edit cheevos from other sets from other developers
-Edit/Change rich presences, leaderboards, gameicon, images, box art, game details from other sets from other developers
-Unlink/Rename game from other sets from other developers and **his own sets** too
-Add/Remove game ID from other sets from other developers and **his own sets** too
-Resolve/Close/Reopen tickets from other sets from other developers

**Comment by meleu** [comment]
I'm afraid we are derailing a bit...

Let me try to summarize the discussion up to now:
- @Blazekickn requested to allow all registered users to be able to view code notes because it can help jr-devs.
- @kdecks stated that allowing all users to see Code Notes can make things easier for cheaters.
- as a reasonable mid-term between the original request and kdecks statement, @meleu and @luchaos are implementing a new role for jr-devs and allow them to see the Code Notes (plus the achievement's logic). The implementation is relatively simple and do not require to much work on a code base that is about to be deprecated.

@steffeN-92 your ideas are cool, indeed. But those sophisticated solutions are not going to be implemented on v1. Please, open a new issue with your feature request (suggestion for title: "permissions for jr-devs on RAWeb v2"). The staff was talking about roles/permissions sometime ago (mainly @kdecks and @luchaos) and I'm sure your suggestions, if not already present, can be added to the doc we are using to list each role's permissions.

**Comment by Jamiras** [comment]
The original ask, to allow code notes to be accessed by any registered user, was implemented in #884. 

I think everything else here is referring to repurposing Super User to be Jr. Dev, which was implemented in #617.

I believe this can be closed (or recategorized as it's not really possible to close a discussion without deleting it).
